tools/scripts/bumpenvs.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python3
+
+"""
+Update all the environment image tags in the repository based on bumpenvs.yaml.
+
+Usage: bumpenvs.py path/to/bumpenvs.yaml
+"""
+
+import os
+import subprocess
+import sys
+from typing import List
+
+import yaml
+
+
+def samefile(a: str, b: str) -> bool:
+    return os.path.samefile(os.path.expanduser(a), os.path.expanduser(b))
+
+
+def list_git_paths_with_tag(tag: str, ignore_files: List[str]) -> List[str]:
+    cmd = ["git", "rev-parse", "--show-toplevel"]
+    r = subprocess.run(cmd, stdout=subprocess.PIPE, universal_newlines=True, check=True)
+
+    git_root = r.stdout.strip("\n")
+
+    cmd = ["git", "grep", "-z", "-l", tag, git_root]
+    r = subprocess.run(cmd, stdout=subprocess.PIPE, universal_newlines=True, check=True)
+    paths = r.stdout.strip("\0\n").split("\0")
```

Comment:
non-blocking: I'm not seeing any extra `\n`s in `-z` mode (that would defeat the whole point of `-z`, no?). Also, this can more precisely be `rstrip` instead of `strip` (also the call above), which is quite unlikely to make an actual difference but would be more faithful to the desired effect.

Reply:
Hm.  When I run `git grep -z -l` interactively I do get one newline at the end of everything, but when I pipe the output or run it as a subprocess I don't.  Must be some `isatty` magic or something.  I'll take out the `\n`.